src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
```

Comment:
PullProofAI

There is a potential security vulnerability in the current implementation of the `FeedbackSecretGuard` class. The comparison of the `requestSecret` with the `secret` is done using strict equality (`===`), which might not be secure enough for sensitive data like secrets.

#### Recommendation
Consider using a secure comparison method such as a constant-time comparison function to mitigate timing attacks. You can use libraries like `crypto.timingSafeEqual` in Node.js to perform secure comparisons.

```diff
-     if (requestSecret && requestSecret === secret) {
+     if (requestSecret && secureCompare(requestSecret, secret)) {
```

By implementing a secure comparison method, you can enhance the security of the secret validation process and reduce the risk of timing attacks.


<!-- This is an auto-generated comment by PullProofAI -->